File: packages/gbnf-experiment/data/20260909T130706Z_31ac2789/ported_implementation/src/utils/is_point_in_range.ts
import type { Range } from '../grammar_graph/grammar_graph_types.ts';

export const isPointInRange = (point: number, givenRange: Range): boolean => {
  if (!Number.isInteger(point)) {
    throw new Error('point must be an integer');
  }
  return point >= givenRange[0] && point <= givenRange[1];
};

export const is_point_in_range = isPointInRange;

File: proposed-projects/round-trip-experiment/reverse/20260909T151143Z_170b737d/20260910T085124Z_7f672549/ported_implementation/gbnf/utils/is_point_in_range.py
from ..grammar_graph.grammar_graph_types import Range


def is_point_in_range(point: int, given_range: Range) -> bool:
    if not isinstance(point, int) or isinstance(point, bool):
        raise ValueError("point must be an integer")
    return given_range[0] <= point <= given_range[1]
